src/expr/macro/src/context.rs

```diff
@@ -138,7 +138,47 @@ pub(super) fn generate_captured_function(
     mut user_fn: ItemFn,
 ) -> Result<TokenStream> {
     let CaptureContextAttr { captures } = attr;
-    let orig_user_fn = user_fn.clone();
+    let is_async = user_fn.sig.asyncness.is_some();
+    let mut orig_user_fn = user_fn.clone();
+    if is_async {
+        // Modify the return type to impl Future<Output = output> + Send + 'static for the original function.
+        let output_type = match &orig_user_fn.sig.output {
+            ReturnType::Type(_, ty) => ty.clone(),
+            ReturnType::Default => Box::new(syn::parse_quote!(())),
+        };
+        orig_user_fn.sig.output = ReturnType::Type(
+            syn::token::RArrow::default(),
+            Box::new(
+                syn::parse_quote!(impl std::future::Future<Output = #output_type> + Send + 'static),
+            ),
+        );
+        orig_user_fn.sig.asyncness = None;
```

Comment:
Curious: why is it necessary to build a non-async function with `async move {...}` instead, instead of just building a `async fn`?

Reply:
Currently, `cargo expand` shows

```rs
fn pg_is_in_recovery_impl(
    meta_client: &Arc<dyn FrontendMetaClient>,
) -> impl std::future::Future<Output = Result<bool>> + Send + 'static {
    let meta_client = meta_client.clone();
    async move {
        { Ok(meta_client.check_cluster_in_recovery().await.unwrap_or(true)) }
    }
}
```

I was asking why not 

```rs
async fn pg_is_in_recovery_impl(meta_client: &Arc<dyn FrontendMetaClient>) -> Result<bool> {
    let meta_client = meta_client.clone();
    Ok(meta_client.check_cluster_in_recovery().await.unwrap_or(true))
}
```

Reply:
Because the `try_with` function in local task only accept non-asynchronous lambda functions.